app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
```

Comment:
Take a look at pagination/limiting of Pictures so you don't potentially load every picture in the DB (which could be 1000s!)